src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -786,13 +784,14 @@ Slave waitForSlaveToComeOnline(Pod pod, String namespace) throws Exception {
          * @return exception to throw
          */
         protected IllegalStateException failureToConnectOnlineSlave(Pod pod, int slaveConnectTimeoutInSeconds) {
-            return new IllegalStateException(format("Slave is not connected and online after %d seconds, status: %s", slaveConnectTimeoutInSeconds, getPodStatus(pod)));
+            return new IllegalStateException(format("Slave is not connected and online after %d seconds, status: %s",
+                    slaveConnectTimeoutInSeconds, getPodStatus(pod, true)));
```

Comment:
Looks like this is the only place in code, where `getPodStatus` is called with `true`. Therefore I would suggest calling `pod = getPodByNamespaceAndPodId(getPodNameSpace(pod), getPodId(pod));` right here, just before checking pod status, instead of adding a flag to `getPodStatus` method signature.